### ISSUE TITLE
Any way to input / interact into websites on Chromecast?

### ISSUE THREAD

**Issue by weirdroid** [skorokithakis/catt#168]
I am successfully using CATT to display a private security camera web UI on a Chromecast device.
However I want to require authentication, otherwise it will be available without auth on the web.  Also may want to interact with the UI.
There must be many other websites played by CATT users that require input and/or interaction, right?
How do you do it? 
Is there any remote / virtual keyboard, or a way to connect a keyboard to allow me to input a username / password into a site running in a Chromecast device?
I have searched like hell and tried chatting with Google support, to no avail.

**Comment by weirdroid** [comment]
Thank you for your feedback.
I will try that, but my understanding is Chromecast will only display URLs
that are officially signed CA certified domains using encryption with HTTPS.
I tried using the URL of the stream with the video player VLC, which is
Chromecast enabled, but found that program to be mostly unable to display
may IP camera's RTSP streams.

Thanks again and I will let the forum know how it goes.

**Comment by weirdroid** [comment]
Thank you, that's great information..

**Comment by weirdroid** [comment]
I am quite confused, as I learned that Chrome (therefore Chromecast) no
longer allows including the auth in the URL to comply with security
standards.
Are you saying that embedding the user:password in the URL is still working
for you today?

I changed my camera server (Blue Iris) to allow basic auth and tried *catt
cast_site*  to display web access by embedding the auth in the URL and got
*very* close.
It actually signs in and starts showing the page but then breaks down and
displays an error about the URL being illegal.

I can succeed by setting my Blue Iris to allow *no* auth on LAN-only
connections.
Then using *catt cast_site* will work and display the website.
But that doesn't work for me because even outside WAN connections appear as
LAN connections to my server, so the website would be open from anywhere.
I think this is due to the HTTPS proxy I have in front of the web server.

Incidentally, the reason for the HTTPS proxy (stunnel
<https://www.stunnel.org/>) with a CA-signed encryption certificate (
LetsEncrypt <https://letsencrypt.org/>) was to get Chromecast to work from
the Android app for Blue Iris.
This caused me to believe that Chromecast will only accept a website with a
recognized HTTPS domain, which I also learned is a recent change by Google
for security reasons.
You must have deep understanding of Chormecast -- is that true about only
accepting websites with certified HTTPS domains?

I have had no luck yet getting Chomecast to display any direct URL streams
(from cameras in my case) as you suggest, but I will keep thinking and
trying.
